File: sdkwork-ai-sdk-csharp/Api/ResponsesApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class ResponsesApi
    {
        private readonly HttpClient _client;

        public ResponsesApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Cancel response
        /// </summary>
        public async Task<CancelResponseResponse?> CancelResponseAsync(string response_id)
        {
            return await _client.PostAsync<CancelResponseResponse>(ApiPaths.AiPath($"/responses/{response_id}/cancel"), null);
        }

        /// <summary>
        /// Create response
        /// </summary>
        public async Task<CreateResponseResponse?> CreateResponseAsync(CreateResponseRequest body)
        {
            return await _client.PostAsync<CreateResponseResponse>(ApiPaths.AiPath("/responses"), body);
        }

        /// <summary>
        /// List response input items
        /// </summary>
        public async Task<ListResponseInputItemsResponse?> ListResponseInputItemsAsync(string response_id)
        {
            return await _client.GetAsync<ListResponseInputItemsResponse>(ApiPaths.AiPath($"/responses/{response_id}/input_items"));
        }

        /// <summary>
        /// Get response
        /// </summary>
        public async Task<GetResponseResponse?> GetResponseAsync(string response_id)
        {
            return await _client.GetAsync<GetResponseResponse>(ApiPaths.AiPath($"/responses/{response_id}"));
        }

        /// <summary>
        /// Delete response
        /// </summary>
        public async Task<DeleteResponseResponse?> DeleteResponseAsync(string response_id)
        {
            return await _client.DeleteAsync<DeleteResponseResponse>(ApiPaths.AiPath($"/responses/{response_id}"));
        }
    }
}

File: sdkwork-ai-sdk-swift/Sources/API/ResponsesApi.swift
import Foundation

public class ResponsesApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Cancel response
    public func cancelResponse(response_id: String) async throws -> CancelResponseResponse? {
        let response = try await client.post(ApiPaths.aiPath("/responses/\(response_id)/cancel"), body: nil)
        return response as? CancelResponseResponse
    }

    /// Create response
    public func createResponse(body: CreateResponseRequest) async throws -> CreateResponseResponse? {
        let response = try await client.post(ApiPaths.aiPath("/responses"), body: body)
        return response as? CreateResponseResponse
    }

    /// List response input items
    public func listResponseInputItems(response_id: String) async throws -> ListResponseInputItemsResponse? {
        let response = try await client.get(ApiPaths.aiPath("/responses/\(response_id)/input_items"))
        return response as? ListResponseInputItemsResponse
    }

    /// Get response
    public func getResponse(response_id: String) async throws -> GetResponseResponse? {
        let response = try await client.get(ApiPaths.aiPath("/responses/\(response_id)"))
        return response as? GetResponseResponse
    }

    /// Delete response
    public func deleteResponse(response_id: String) async throws -> DeleteResponseResponse? {
        let response = try await client.delete(ApiPaths.aiPath("/responses/\(response_id)"))
        return response as? DeleteResponseResponse
    }
}
